### PR TITLE
Add password reset policy 464

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* generate filter handler bean based on property.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CEntryPoint.java [hunk 1]
```diff
@@ -41,9 +41,8 @@ public void commence(HttpServletRequest request, HttpServletResponse response, A
         }
 
         final String redirectURL = AADB2CURL.getOpenIdSignUpOrSignInURL(b2cProperties, requestURL, request);
+        redirectStrategy.sendRedirect(request, response, redirectURL);
 
         log.debug("Authentication is required to access URL {}.", requestURL);
-
-        redirectStrategy.sendRedirect(request, response, redirectURL);
```
Comment (xscript) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CEntryPoint.java]: Why do we change the order here?

Comment (Incarnation-p-lee) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CEntryPoint.java]: Yes, should not change this, will revert this.

Comment (xscript) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CEntryPoint.java]: Also let's refine the debug log to `Authentication is required to access URL {}. Redirecting to {}`.

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CProperties.java [hunk 2]
```diff
@@ -61,11 +83,17 @@
     public static class Policies {
 
         /**
-         * The sign-up-or-sign-in policies that created under b2c tenant.
+         * The sign-up-or-sign-in policy that created under b2c tenant.
```
Comment (xscript) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CProperties.java]: nit: -> `...policy which is created...`

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CProperties.java [hunk 3]
```diff
@@ -61,11 +83,17 @@
...
          */
         @JsonProperty(SIGN_UP_OR_SIGN_IN)
         private Policy signUpOrSignIn = new Policy();
 
+        /**
+         * The password-reset policy that created under b2c tenant.
```
Comment (xscript) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CProperties.java]: nit: -> `... policy which is created ...`

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CLogoutSuccessHandler.java [hunk 4]
```diff
@@ -29,33 +27,12 @@ public AADB2CLogoutSuccessHandler(@NonNull AADB2CProperties b2cProperties) {
...
-        this.logoutSuccessURL = logoutSuccessURL;
-
-        return this;
-    }
-
     @Override
     public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication)
             throws IOException {
-        final String redirectURL = AADB2CURL.getOpenIdLogoutURL(b2cProperties, getLogoutSuccessURL(), request);
+        redirectStrategy.sendRedirect(request, response, AADB2CURL.getOpenIdLogoutURL(b2cProperties, request));
```
Comment (xscript) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CLogoutSuccessHandler.java]: Let's also keep the order here.

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CFilterScenarioHandler.java [hunk 5]
```diff
@@ -16,25 +17,32 @@
...
  * ${@link AADB2CFilterDefaultHandler} is default handler for AAD B2C filter, do nothing at all.
  * <p>
- * ${@link AADB2CFilterSignUpOrInHandler} will obtain the result from AAD B2C, it will generate UserPrinciple
+ * ${@link AADB2CFilterPolicyReplyHandler} will obtain the result from AAD B2C, it will generate UserPrinciple
  * and ${@link org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken} for
  * spring security context if success, or take care the error response and throw exception.
  * <p>
- * ${@link AADB2CFilterLogoutSuccessHandler} will obtain the result from AAD B2C, it will do nothing if success,
- * or warn the user that the token is expired.
+ * ${@link AADB2CFilterPasswordResetHandler} will take care the password reset url and redirect to AAD B2C for
```
Comment (xscript) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CFilterScenarioHandler.java]: nit: -> `take care of`

**Comment by Incarnation-p-lee** [comment]
Resolved java doc and log message.

**Pull request merged**